substrate/frame/election-provider-multi-phase/src/signed.rs

```diff
@@ -509,6 +509,11 @@ impl<T: Config> Pallet<T> {
 		let positive_imbalance =
 			T::Currency::deposit_creating(who, reward.saturating_add(call_fee));
 		T::RewardHandler::on_unbalanced(positive_imbalance);
+
+		// register submitter as whitelisted for the next round(s).
+		SignedWhitelist::<T>::mutate(|w| {
+			w.force_push(who.clone());
```

Comment:
Could you add a FIFO test? I tried and does not work as expected.
This is what I got:

```
whitelist(1); // BoundedVec([1], 2)
whitelist(2); // BoundedVec([1, 2], 2) 
whitelist(3); // BoundedVec([1, 3], 2)
```

Reply:
I have fixed the code and added a test to ensure we have a FIFO mechanism in place. It should be good now!